### PR TITLE
Adding basic js setup, gitignore, eslint

### PR DESCRIPTION
**Pull request by fiddur**
* Adding eslint config (very basic, not much style since I don't know your code style in detail and it isn't consistent regarding trailing comma in multiline object for example).
* Running eslint before tests.
* Adding package-lock.json since the versions should be locked down, especially in a security sensitive environment.
* Adding gitignore of node_modules.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  parserOptions: {
+    ecmaVersion: 6,
+  },
+  env: {
+    node: true,
+    mocha: true,
+  },
+  'extends': 'eslint:recommended',
```
Comment (TMSCH) [inline .eslintrc.js]: You could use the Google JS ESLint config: https://github.com/google/eslint-config-google, maybe [this config](https://github.com/google/eslint-config-google#using-the-google-config-with-eslintrecommended).

**Review by TMSCH — CHANGES_REQUESTED** [review]
Thanks for adding this! Let me know what you think re: my comment.

File: .eslintrc.js (hunk 1 above)
Comment (fiddur) [reply .eslintrc.js]: Ah great, you have one ready :)

**Comment by fiddur** [comment]
I suggest following this up with a separate PR just doing `eslint --fix src test`; at least 264 of the linting errors can be automatically fixed :)

**Review by TMSCH — APPROVED** [review]
Sounds good!

**Pull request closed without merging**